src/engine/specification.md

```diff
@@ -174,6 +180,25 @@ This structure maps on the [`ExecutionPayload`](https://github.com/ethereum/cons
 - `blockHash`: `DATA`, 32 Bytes
 - `transactions`: `Array of DATA` - Array of transaction objects, each object is a byte list (`DATA`) representing `TransactionType || TransactionPayload` or `LegacyTransaction` as defined in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)
 
+### ExecutionPayloadHeaderV1
+
```

Comment:
I don't think a full `ExecutionPayloadHeader` is required to trigger the beacon sync, only  `blockNumber` and `blockHash` are sufficient and rest EL clients can fetch off the wire, since they would anyway need to fetch full block body for `transactions` to eventually execute the block (and if we are being liberal we can include `parentHash` as well)
So may be we can call it `ExecutionPayloadHeaderSlim` or something

Reply:
For LES use case, having access to `logs_bloom` is also useful to determine whether a block contains useful information.

The one truly questionable field is the `transactionsRoot` though – it being an SSZ root may make it difficult to get a `VALID` / `INVALID` classification from the EL without it also knowing how to create and verify those. For that one, I see three options:
1. `transactionsRoot` is not useful for any use case, in this case just drop it from the engine API structure, diverging from the beacon chain structure.
2. `transactionsRoot` would be useful as an RLP hash for some use cases, in this case extend the beacon chain structure with RLP `transactionsHash` that lives next to the `transactions` in both `ExecutionPayload` and `ExecutionPayloadHeader`.
3. `transactionsRoot` is still useful in SSZ format, in this case just keep as is. Maybe for MEV blinded blocks to prove inclusion of particular tx against censoring, but those proofs could also be rooted in `stateRoot` I guess?

On CL light client front, the extra size for full header is not substantially different from including just a couple of the fields, as proof size goes up for individual fields. So, full `ExecutionPayloadHeader` can be expected to be available. Any subset is alright for the engine API.

Reply:
Makes sense, so...

for execution nodes running `full` protocol, it won't matter what kind of `transactionsRoot` we provide since they would still need to download the block off the wire (as you observed) using their own peers as they do in normal beacon sync when they don't have parents.

Now for `les` we could make it easy by having the RLP transactions hash  that the blockhash lines up and they don't have to download anything off wire (unless there is a missing header update in which case they will again have to)

Reply:
https://github.com/ethereum/consensus-specs/pull/3078
Opened a proposal here to add those RLP hashes to the `ExecutionPayload` / `ExecutionPayloadHeader` as well, from Capella onward. This would make this much easier, removing the need for RLP computations in CL, or for SSZ computations in EL.